### PR TITLE
	modified:   gumby/scenario.py

### LINKED ISSUE #499: Suggestion: remove for-loops from the scenario file scripts
Currently, Gumby scenario files have support [for for loops](https://github.com/Tribler/gumby/blob/devel/gumby/scenario.py#L148). This feature, however, is not used currently and has never been used. As such, I suggest to remove it.

Estimated time: less than a day.

### PR DESCRIPTION
**Pull request by pvanita**
for_loops code has been removed from scenario.py

Fixes #499

### CODE REVIEW

File: gumby/scenario.py [hunk 1]
```diff
@@ -145,47 +145,47 @@ def _parse_arguments(self, args):
...
-
-        # The lower and upper bounds are considered inclusive, hence the additional in/decrements
-        if lo_bound > hi_bound:
-            offset = -1
-            hi_bound -= 1
-        else:
-            hi_bound += 1
-
-        return callable_function, args, lo_bound, hi_bound, offset, control_var
+##    def _parse_for_loop(self, loop):
```
Comment (devos50) [inline gumby/scenario.py]: Instead of commenting out the code, please remove it 👍

**Review by devos50 — CHANGES_REQUESTED** [review]

**Comment by devos50** [comment]
ok to test

File: gumby/scenario.py (hunk 1 above)
Comment (pvanita) [reply gumby/scenario.py]: Thanks Martijn, I will do that asap

Comment (devos50) [reply gumby/scenario.py]: In addition, there are also some tests related to for-loops. You can also remove them.

Comment (pvanita) [reply gumby/scenario.py]: Only 1 for-loops related test has been detected and removed ( file: gumby/tests/scenario_language )

**Comment by pvanita** [comment]
For_loop code has actually removed and not just commented out. 
Test file for for-loops has been removed ( removed file: gumby/tests/scenario_language )

**Comment by synctext** [comment]
@devos50 Could you possibly repeat the open issues with this PR please?

**Comment by devos50** [comment]
@synctext what do you mean by 'repeating the open issues'? 🤔

**Comment by devos50** [comment]
@pvanita please note that you can request a review from me by clicking the 're-request review' button in the top-right part of this page:

Schermafbeelding 2021-12-16 om 11 16 15

**Review by devos50 — APPROVED** [review]
Looks good! 👍

**Review by devos50 — CHANGES_REQUESTED** [review]
In most of our projects, we try to spend some time on writing a consistent commit history. This includes writing meaningful and clear commit messages, which helps us to understand why a particular change has been made.

Instead of the current commit history of this PR (which contains five commits that state which files have been modified), I would like to see a cleaner commit history. In this case, I think it's fine to squash all your changes to a single commit. This commit should then be named something like: "Removed support for for-loops", clearly indicating which change the commit contains. Sometimes it can happen that you make changes that are not related to for-loops in the same PR - these changes should go in a a separate commit (but that does not apply to this PR). More information on writing commit messages can be found [here](https://cbea.ms/git-commit/).

Please cleanup your commit history and I will merge this PR 👍

**Comment by pvanita** [comment]
The commit history for the RemoveScenarioLoop branch has been cleaned up

**Review by devos50 — APPROVED** [review]
Looks good, thank you! 👍 

Tip for next time: please link any issue that is relevant in the first message of this PR (or even in the commit message), e.g., "fixes #499". This will make sure that the linked issue will automatically be closed when this PR is merged.

**Comment by pvanita** [comment]
Thank you for the advice Devos50, I will link the issues for all future PR's

**Pull request merged**